mlir/include/mlir/Interfaces/ValueBoundsOpInterface.h

```diff
@@ -198,6 +198,28 @@ class ValueBoundsConstraintSet {
                        std::optional<int64_t> dim1 = std::nullopt,
                        std::optional<int64_t> dim2 = std::nullopt);
 
+  /// Traverse the IR starting from the given value/dim and add populate
+  /// constraints as long as the currently set stop condition holds. Also
+  /// processes all values/dims that are already on the worklist.
+  void populateConstraints(Value value, std::optional<int64_t> dim);
```

Comment:
Some of this may clash with #83876. I'm going to rebase this PR when #83876 has been merged.

Reply:
Thanks! I'll land it soon :)

Reply:
I rebased all PRs. It would be great if you could also review #86097, then I can start merging PRs. You probably know the codebase best out of all reviewers that I added to that PR.